File: ReversePyramid.py
class ReversePyramid:
    def __init__(self, rows):
        self.rows = rows

    def printPyramid(self):
        for i in range(1, self.rows + 1):
            for j in range(1, i):
                print(" ", end="")
            for j in range(self.rows - i, -1, -1):
                print("*", end="")
            for j in range(self.rows - i, 0, -1):
                print("*", end="")
            print()


class ChildReversePyramid(ReversePyramid):
    def __init__(self, r):
        super().__init__(r)


if __name__ == '__main__':
    row = int(input("Enter the number of rows: "))
    print()
    obj = ChildReversePyramid(row)
    obj.printPyramid()

File: ReversePyramid.cpp
#include <iostream>

using namespace std;


class ReversePyramid {
    private:
        int rows;
    public:
        ReversePyramid(int r) {
            rows = r;
        }
    
    void printPyramid() {
        for(int i=1;i<=rows;i++){
            for(int j=1;j<i;j++){
                cout<<" ";
            }
            for(int j=rows-i;j>=0;j--){
                cout<<"*";
            }
            for(int j=rows-i;j>0;j--){
                cout<<"*";
            }
            cout<<"\n";
        }
    }
    
    ~ReversePyramid(){
    }
};

class ChildReversePyramid : public ReversePyramid {
    public:
        ChildReversePyramid(int r) : ReversePyramid(r){
            
        }
};
int main()
{
    int ro;
    cout<<"Enter the number of rows: ";cin>>ro;
    cout<<"\n";
    ChildReversePyramid obj(ro);
    obj.printPyramid();
    return 0;
}
